Content.Server/GameObjects/Components/Construction/RefinableComponent.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Threading.Tasks;
+using Content.Server.GameObjects.Components.Interactable;
+using Content.Server.GameObjects.Components.Stack;
+using Content.Shared.GameObjects.Components.Interactable;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Construction
+{
+    /// <summary>
+    /// Used for something that can be refined by welder.
+    /// For example, glass shard can be refined to glass sheet.
+    /// </summary>
+    [RegisterComponent]
+    public class RefinableComponent : Component, IInteractUsing
```

Comment:
I think you should come up with a less generic name than this, since it only works with welders and there may be other, more generic kinds of refining added later. Maybe `WelderRefinableComponent`?